sclack/components.py

```diff
@@ -564,13 +564,32 @@ def keypress(self, size, key):
         return super(MessagePrompt, self).keypress(size, key)
 
 class Profile(urwid.Text):
-    def __init__(self, name, is_online=False):
+    def __init__(self, name, is_online=False, is_snoozed=False):
+        self.name = name
+        self.is_snoozed = is_snoozed
+
         if is_online:
-            presence_icon = ('presence_active', ' {} '.format(get_icon('online')))
+            self.presence_icon = ('presence_active', ' {} '.format(get_icon('online')))
         else:
-            presence_icon = ('presence_away', ' {} '.format(get_icon('offline')))
-        body = [presence_icon, name]
-        super(Profile, self).__init__(body)
+            self.presence_icon = ('presence_away', ' {} '.format(get_icon('offline')))
+
+        super(Profile, self).__init__(self.body)
+
+    @property
+    def body(self):
+        return [self.presence_icon, self.name, self.snooze_indicator]
+
+    @property
+    def snooze_indicator(self):
+        if self.is_snoozed:
+            return ' (snoozed) '
+        else:
+            return ''
+
+    def change_snooze_indicator(self, is_snoozed=None):
```

Comment:
I think `is_snoozed=False`, am I correct?

Reply:
Also change status indicator icon when is_snoozed is changed

Reply:
> I think `is_snoozed=False`, am I correct?

I meant to set a snooze to `True` or `False` only if someone explicitly said so. It was just a semantic idea. But yeah, it could be only `is_snooze=False` (and then I'd remove the `if` below).

> Also change status indicator icon when is_snoozed is changed

That led me to the other thread's question: should this snooze be a separated indicator or just be the same as `presence_icon`?

Reply:
I think this indicator should be only one, but we have some statuses
* Snooze
* Online

Reply:
I tried to update it slightly, please check https://i.imgur.com/at0XuwE.png